### PR TITLE
KOGITO-2485 Share KafkaContainer to be used by Example tests

### PR DESCRIPTION
**Pull request by Sgitario**
JIRA: https://issues.redhat.com/browse/KOGITO-2485
Description:
- Share the same Kafka Container to be used by the Kogito Example tests: this will allow to set the configuration only in one place, so all the tests will automatically be updated.
- Configure the process-springboot-example and process-quarkus-example with this new Kafka container, so we remove the external dependency of Kafka when running mvn clean install -Pevents.

### CODE REVIEW

File: examples-test-utils/pom.xml [hunk 1]
```diff
@@ -34,11 +34,23 @@
...
     <dependency>
       <groupId>io.quarkus</groupId>
       <artifactId>quarkus-junit5</artifactId>
       <scope>provided</scope>
     </dependency>
+    
+    <dependency>
+      <groupId>org.apache.kafka</groupId>
+      <artifactId>kafka-clients</artifactId>
+      <scope>provided</scope>
```
Comment (Sgitario) [inline examples-test-utils/pom.xml:50]: As not all the examples might require the KafkaClient, I set this dependency as provided.

File: examples-test-utils/src/main/java/org/kie/kogito/testcontainers/KogitoKafkaContainer.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+/**
+ * Kafka Container for Kogito examples.
+ */
+public class KogitoKafkaContainer extends KafkaContainer implements ConditionalTestResource<KogitoKafkaContainer> {
+
+    public static final String QUARKUS_KAFKA_BOOTSTRAP_SERVERS = "kafka.bootstrap.servers";
+    public static final String SPRINGBOOT_KAFKA_BOOTSTRAP_SERVERS = "spring.kafka.bootstrap-servers";
+
+    private static final String CLOUD_EVENTS_PROPERTY = "tests.cloud.events";
```
Comment (Sgitario) [inline examples-test-utils/src/main/java/org/kie/kogito/testcontainers/KogitoKafkaContainer.java]: Some examples need to run the persistence and events at the same time. That's why we can't use the default test.category for this.

**Comment by Sgitario** [comment]
I ran all the examples with no external dependency of Kafka or Infinispan, just running:
mvn -fn clean install and mvn -fn clean install -Pevents
The only failing example is "dmn-drools-quarkus-metrics", but it's working fine running in the pipeline, so I'm not worry about it.

**Review by sutaakar — APPROVED** [review]

**Review by radtriste — APPROVED** [review]

**Comment by Sgitario** [comment]
jenkins please retest this

**Comment by Sgitario** [comment]
jenkins please retest this

File: examples-test-utils/pom.xml [hunk 3]
```diff
@@ -34,11 +34,23 @@
       <scope>compile</scope>
     </dependency>
     
+    <dependency>
+      <groupId>org.testcontainers</groupId>
+      <artifactId>kafka</artifactId>
+      <scope>compile</scope>
```
Comment (cristianonicolai) [inline examples-test-utils/pom.xml]: compile is the default anyway, doesn't need to be specified.

**Review by cristianonicolai — COMMENTED** [review]
one minor comment

**Review by cristianonicolai — APPROVED** [review]

File: examples-test-utils/pom.xml (hunk 3 above)
Comment (Sgitario) [reply examples-test-utils/pom.xml]: Removed

**Comment by Sgitario** [comment]
@jstastny-cz do you mind taking a look into this PR? Thanks!

**Comment by Sgitario** [comment]
jenkins please retest this

**Comment by Sgitario** [comment]
jenkins please retest this

**Review by jstastny-cz — APPROVED** [review]
LGTM, thanks.

**Pull request merged**